dlib/dnn/trainer.h

```diff
@@ -1198,15 +1198,18 @@ namespace dlib
             job.test_only = test_only;
 
             // chop the data into devs blocks, each of about block_size elements.
-            size_t block_size = (num+devs-1)/devs;
+            const double block_size = num / static_cast<double>(devs);
```

Comment:
How about adding something like this as well?
```
CASSERT(num >= devs, "You need to give at least one training sample for each device you want to use.");
```

Reply:
I was thinking about that, too... But I thought it's not even necessary in the generic case (when not using batch normalization, for example)?

Reply:
Yeah, on second thought if someone wants to do it we should let them.